### ISSUE TITLE
Imported Nmap Scan - Service Version Info Shown Incorrectly

### ISSUE THREAD

**Issue by xpelican** [GoVanguard/legion#164]
_Using Legion 0.3.6-1580902879 on Ubuntu 18.04_

Hello,

After importing an nmap scan, one service that nmap reported as OpenSSH **5.3** is getting displayed in Legion as OpenSSH **7.4**

Unfortunately I cannot share the nmap scans for reproduction as they contain private info. But there were a lot of actual 7.4s in the network and only one instance of 5.3, so maybe that somehow caused this result?

In any case, this is a big problem as it means a user cannot fully trust what Legion shows.

Let me know if I can help in any way with this other than sharing the scan files.

Thanks in advance!

**Comment by jchoy14** [comment]
We're still trying to replicate this, but unfortunately we haven't been able to yet. Would you be able to import those scans into Sparta and see if the same issue presents itself?

**Comment by xpelican** [comment]
Good idea. Just tried it, and no, this problem doesn't happen with Sparta (latest release on GitHub), only in Legion.

**Comment by mcjon3z** [comment]
I've been noticing the same issue. Here are screenshots of same machine in Zenmap and the related Legion after import:

<img width="501" alt="image" src="https://user-images.githubusercontent.com/6471785/77437210-c4b8a000-6dba-11ea-8094-d72bcfec7621.png">

<img width="679" alt="image" src="https://user-images.githubusercontent.com/6471785/77437321-e6b22280-6dba-11ea-9431-e8e8086c0995.png">

**Comment by mcjon3z** [comment]
Just observed the same behavior when running a scope scan with custom parameters including -sV instead of the default staged scan. In the scan scope above, the staged scan did not report any version information at all for http services. The custom scan reported Microsoft IIS httpd 10.0 for every http service identified.

**Comment by mcjon3z** [comment]
Did some more test scans once I got home and it looks like it is just grabbing the service info from the first existing entry and duplicating it. Here is a screenshot of the SSH service detected on 3 of my machines (the first is an iLO, the others are linux boxes):

<img width="764" alt="image" src="https://user-images.githubusercontent.com/6471785/77498399-ed29b400-6e25-11ea-97d9-663db10693ea.png">

I'll be happy to do some more testing for you guys but I don't have Legion running natively on my main dev machine to do any debugging right now to help track down where the logic error is.

**Comment by jchoy14** [comment]
Thanks all, your feedback is very helpful in helping us pinpoint the issue.

**Comment by Ferrariman1996** [comment]
Greeting,
I have found a difference in Legion's Nmap parameters to Sparta's. Legion uses the parameter **-sC** and Sparta uses **-sV**. I investigated this becuase of my Honeypot paper for my masters degree and the -sV parameter is able to detect Dianaea's (that is a honeypot) fake MS-SQL server, but the -sC detects the version just as a MS-SQL server (tested on same target machine). May I ask what was the reason in Legion to switch to the -sC parameter? Thank you very much for your answer, It would really help me in my paper.
